dts/bindings/i2c/i2c.yaml

```diff
@@ -27,6 +27,11 @@ properties:
       type: int
       category: required
       description: should be 0.
+    current-speed:
+      type: int
+      category: optional
+      description: Initial bus speed in Hz
```

Comment:
what is this suppose to me, or how does it differ from clock-frequency?

Reply:
The UART peripheral currently uses clock-frequency to describe the frequency of the clock driving the peripheral and the current-speed property to describe the baud rate of the output.

Similarly, the I2C peripheral is driven by a clock at a certain frequency and drives the output bus at a certain baud, so we should use the same properties as the UART in the same way:
- clock-frequency describes the input clock to the peripheral
- current-speed describes the configured speed of the I2C bus

Reply:
@galak I looked into how Zephyr's DTS usage compares to Linux's.

Zephyr's use of `clock-frequency` to describe the bus frequency of the I2C bus [is consistent with Linux](https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/i2c/i2c.txt). It's also true that both Zephyr and Linux's UART DTS bindings use `clock-frequency` to describe the input clock rate to the UART peripheral.

Unfortunately this means that I2C and UART use the same property name to describe different things.

I can change this PR to use clock-frequency to describe the I2C bus rate. Do you have any suggestions for how to specify the input clock rate to the I2C peripheral?

Reply:
> I can change this PR to use clock-frequency to describe the I2C bus rate. Do you have any suggestions for how to specify the input clock rate to the I2C peripheral?

At some point we'd have clock nodes and we'd utilize those to figure out such information.  Not sure how far you want to go right now with this from a DT point of view.

Reply:
Up until recently this would have been a `#define` in `board.h`. Since we're putting all those clock rates into the board's .dts file, it makes sense to have some kind of DT property, at least as a hold-over until we have clock nodes. Maybe `input-frequency` or namespace it in `zephyr,input-frequency`?

Reply:
I'm ok if we want to introduce a property in the short term til we have proper clock support in DTS, etc.

Do you have clock nodes in your DTS for linux?

Reply:
The I2C peripheral is supported in Linux by the [i2c-ocores driver](opencores,ip-clock-frequency) which uses either `clocks` to refer to a clock node or the `opencores,ip-clock-frequency` property.